qutebrowser/misc/mhtml.py

```diff
@@ -0,0 +1,411 @@
+        data: The content of the stylesheet to scan.
+    """
+    if isinstance(data, str):
+        data = data.encode("utf-8")
+    urls = []
+    for pattern in _CSS_URL_PATTERNS:
+        for match in pattern.finditer(data):
+            url = match.group("url")
+            if url:
+                urls.append(url)
+    return urls
+
+
+MHTMLPolicy = policy.default.clone(linesep="\r\n", max_line_length=0)
+
+
+def _chunked_base64(data, maxlen=76, linesep=b"\r\n"):
+    """Just like b64encode, except that it breaks long lines.
+
+    Args:
+        maxlen: Maximum length of a line, not including the line separator.
+        linesep: Line separator to use as bytes.
+    """
+    encoded = b64encode(data)
+    result = []
+    for i in range(0, len(encoded), maxlen):
+        result.append(encoded[i:i + maxlen])
+    return linesep.join(result)
+
+
+def _rn_quopri(data):
+    """Return a quoted-printable representation of data."""
+    # See RFC 2045 https://tools.ietf.org/html/rfc2045#section-6.7
+    # The stdlib version in the quopri module has inconsistencies with line
+    # endings and breaks up character escapes over multiple lines, which isn't
+    # understood by qute and leads to jumbled text
+    maxlen = 76
+    whitespace = {ord(b"\t"), ord(b" ")}
+    output = []
+    current_line = b""
+    for byte in data:
+        # Literal representation according to (2) and (3)
+        if (ord(b"!") <= byte <= ord(b"<") or ord(b">") <= byte <= ord(b"~")
+                or byte in whitespace):
+            current_line += bytes([byte])
+        else:
+            current_line += b"=" + "{:02X}".format(byte).encode("ascii")
+        if len(current_line) >= maxlen:
+            # We need to account for the = character
+            split = [current_line[:maxlen - 1], current_line[maxlen - 1:]]
+            quoted_pos = split[0].rfind(b"=")
+            if quoted_pos + 2 >= maxlen - 1:
+                split[0], token = split[0][:quoted_pos], split[0][quoted_pos:]
+                split[1] = token + split[1]
+            current_line = split[1]
+            output.append(split[0] + b"=")
+    output.append(current_line)
+    return b"\r\n".join(output)
+
+
+E_NONE = (None, lambda x: x)
+"""No transfer encoding, copy the bytes from input to output"""
+
+E_BASE64 = ("base64", _chunked_base64)
+"""Encode the file using base64 encoding"""
+
+E_QUOPRI = ("quoted-printable", _rn_quopri)
+"""Encode the file using MIME quoted-printable encoding."""
+
+
+class MHTMLWriter(object):
+
+    """A class for outputting multiple files to a MHTML document.
+
+    Attributes:
+        root_content: The root content as bytes.
+        content_location: The url of the page as str.
+        content_type: The MIME-type of the root content as str.
+        _files: Mapping of location->_File struct.
+    """
+
+    BOUNDARY = "---=_qute-" + str(uuid4())
+
+    def __init__(self, root_content=None, content_location=None,
+                 content_type=None):
```

Comment:
They were checked with `is not None` before, as there's probably no reason they should ever be `None` (you always have the URL or the website, and the Content-Type is hardcoded to `text/html`), but when I switched to the `email.mime` stuff, I forgot to add these checks again.